pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -488,3 +494,21 @@ func getNodeIPs(node *api.Node) []string {
 	}
 	return result
 }
+
+func getSuspendProvisionSettingByNodeType(properties []*proto.EntityDTO_EntityProperty) (disableSuspendProvision bool, nodeType string) {
+	disableSuspendProvision = false
+	for _, property := range properties{
+		spot := strings.Contains(property.GetName(), util.EKSCapacityType) && property.GetValue() == util.EKSSpot
+		windows := strings.Contains(property.GetName(), util.NodeLabelOS) && property.GetValue() == util.WindowsOS
+		if spot || windows {
+			if spot {
+				nodeType = "AWS EC2 spot instance"
+			} else if windows {
+				nodeType = "node with Windows OS"
+			}
+			disableSuspendProvision = true
+			return
+		}
+	}
+	return
+}
```

Comment:
missing new line at file end is causing a build failure

Reply:
done